flux-typeck/src/type_env/paths_tree.rs

```diff
@@ -262,6 +262,21 @@ impl PathsTree {
                             _ => panic!("Unsupported Deref: {elem:?} {ty:?}"),
                         }
                     }
+                    PlaceElem::Index(_) => {
+                        let ty = ptr.borrow().expect_owned();
+                        match ty.kind() {
+                            TyKind::Indexed(BaseTy::Array(arr_ty, _), _)
+                            | TyKind::Exists(BaseTy::Array(arr_ty, _), _) => {
```

Comment:
`TyKind::Exists` shouldn't be possible here because the type should be already unpacked. If it is not, then there's a bug somewhere else.